File: client/lmsHub/src/app/interface/person.ts
import { inherits } from 'util';

export class Person<T> {

  userName: string;
  email: string;
  firstName: string;
  lastName: string;
  age: string;
  relationStatus?: boolean;

  constructor(options: {
    userName?: string;
    email?: string;
    firstName?: string;
    lastName?: string;
    age?: string;
    relationStatus?: boolean;
  } = {}) {
    this.userName = options.userName;
    this.email = options.email;
    this.firstName = options.firstName || '';
    this.lastName = options.lastName || '';
    this.age = options.age || '';
    this.relationStatus = options.relationStatus || false;
   }

}

File: src/main/kotlin/com/lmshub/lifestyle/data_classes/Person.kt
package com.lmshub.lifestyle.data_classes

data class Person(var userName: String,
                  var email: String,
                  var firstName: String,
                  var lastName: String,
                  var age: Int,
                  var relationStatus: Boolean?) {

    companion object {
        private val personInfo = arrayOf(
                Person::userName,
                Person::firstName,
                Person::age,
                Person::email,
                Person::relationStatus)
        }

    override fun toString(): String = "User name: $userName, Email: $email, $firstName, $lastName, $age, $relationStatus"
}


data class Project(val projectName: String,
                   val projectType: String,
                   val teamName: String,
                   val members: List<Person>) {
    companion object {
        val teamMembers = arrayListOf<Person>()
    }

    override fun toString(): String = "$teamMembers"
}


